src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/ResourceIdentifier.cs

```diff
@@ -138,10 +138,16 @@ public override string ToString()
 
             AppendIfNotNull(ref resourceId, "/subscriptions/{0}", Subscription);
             AppendIfNotNull(ref resourceId, "/resourceGroups/{0}", ResourceGroupName);
-            AppendIfNotNull(ref resourceId, "/providers/{0}", provider);
-            AppendIfNotNull(ref resourceId, "/{0}", parentAndType);
-            AppendIfNotNull(ref resourceId, "/{0}", ResourceName);
 
+            // Only append parentAndType and ResoruceName when provider is not empty.
+            // Otherwise the url will be invalid.
+            if (!string.IsNullOrEmpty(provider))
```

Comment:
yes we did. one problem that this change is going to address is that when provider is null but resourceName is not null, we still append the ResourceName (without "/providers/"). so for parentAndType and ResourceName, we only append them when they are not null, and when provider is not null.

---
In reply to: [118149831](https://github.com/Azure/azure-powershell/pull/4019#discussion_r118149831) [](ancestors = 118149831)